### ISSUE TITLE
Function with unparenthesized trailing requires-expression always put on a single line

### ISSUE THREAD

**Issue by 976866b7-fc02-4d23-a1c0-197b08fd7c0e** [llvm/llvm-project#48239]
|  |  |
| --- | --- |
| Bugzilla Link | [48895](https://llvm.org/bz48895) |
| Version | trunk |
| OS | Linux |
| CC | @JohelEGP |

## Extended Description 
.clang-format:
```
BasedOnStyle: LLVM
AllowShortFunctionsOnASingleLine: None
```

Input and expected formatted result:
```C++
auto f(int l, int r) requires requires {
  l *r;
}
{
  return l * r;
}
```

Actual output:
```C++
auto f(int l, int r) requires requires {
  l *r;
}
{ return l * r; }
```

Because this works fine as input and actual formatted output:
```C++
auto f(int l, int r) requires requires {
  l *r;
}
{
  return l * r;
  return l * r;
}
```

**Comment by JohelEGP** [comment]
Seems fixed. New actual output:
```C++
auto f(int l, int r)
  requires requires { l *r; }
{
  return l * r;
}
```

**Comment by rymiel** [comment]
Fixed by 9aab0db13fb6d21d1b70247a9b5e4cf916ee1c3a
